qemu/tests/block_hotplug.py

```diff
@@ -87,6 +87,7 @@ def run_sub_test(params, plug_tag):
         plug_tag = "before_plug"
         run_sub_test(params, plug_tag)
 
+        session = vm.wait_for_login(timeout=timeout)
         for num in range(blk_num):
```

Comment:
Do we need close the session before each `test.fail()` and `test.cancel()` in this for block? Like L#101 and L#106

Reply:
@vivianQizhu 
Hi,
I didn't close the session before test.fail and test.cancel,  session.close is called outside for loop which will call test.fail and test.cancel.

```python
session = vm.wait_for_login(timeout=timeout)
for num in range(blk_num):
    ...
    if params.get("need_plug") == "yes":
        disks_before_plug = find_disk(vm, get_disk_cmd)
        devs = vm.devices.images_define_by_params(image_name,
                                                  image_params, 'disk')
        for dev in devs:
            ret = vm.devices.simple_hotplug(dev, vm.monitor)
            if ret[1] is False:
                test.fail("Failed to hotplug device '%s'."
                          "Output:\n%s" % (dev, ret[0]))
    ...
    if params.get("disk_op_cmd"):
        if params.get("os_type") == "linux":
            test_cmd = params.get("disk_op_cmd") % (disk, disk)
        elif params.get("os_type") == "windows":
            test_cmd = params.get("disk_op_cmd") % (disk_letter[num],
                                                    disk_letter[num])
            test_cmd = utils_misc.set_winutils_letter(session, test_cmd)
        else:
            test.cancel("Unsupported OS type '%s'" % params.get("os_type"))
    ...
session.close()
```

Thanks.

Reply:
Maybe I misunderstand your comments.

Reply:
I suppose that `test.fail()` and `test.cancel()` will interrupt the test so it won't go to the outside `session.close()`

Reply:
@YongxueHong I don't think so, `avocado_vt.test.cleanup_env` will only clean `Env.data`, which includes `vms` but not `sessions`, and the sessions opened actually are not property of vms, so I suppose that they will not be cleaned by `Env.destroy()`. It is okay to leave them there since vm does not exist so that the connect will be interrupt anyway, but the session might hang for a long time there before it detect that connection is lost.